### PR TITLE
Added Unit Test Cases for ZimReaderContainerUtils and removing redundant toUri call

### LINKED ISSUE #4918: Add test cases for `ZimReaderContainerUtils` helper class.
See https://app.codecov.io/gh/kiwix/kiwix-android/blob/main/core%2Fsrc%2Fmain%2Fjava%2Forg%2Fkiwix%2Fkiwixmobile%2Fcore%2Futils%2FZimReaderContainerUtils.kt for more details.

### PR DESCRIPTION
**Pull request by atharvyadav22**
Fixes: #4918

### CODE REVIEW

**Comment by atharvyadav22** [comment]
@MohitMaliFtechiz @gouri-panda 
Ready for review.
Thanks

**Comment by MohitMaliFtechiz** [comment]
@atharvyadav22 Thank you. I have just finished the https://github.com/kiwix/kiwix-android/pull/4746. Now i will review all your PRs :) Sorry for the delay.

**Comment by atharvyadav22** [comment]
> @atharvyadav22 Thank you. I have just finished the #4746. Now i will review all your PRs :) Sorry for the delay.

No issues they are not blocker to any of the work, you may have a look when free :)

**Comment by MohitMaliFtechiz** [comment]
> No issues they are not blocker to any of the work

@atharvyadav22 The migration work is very slow. So it needs to be done as a priority. Now, https://github.com/kiwix/kiwix-android/pull/4746 is almost done. The only `CoreReaderFragment` is left to refactor, which is a huge class and will take time. Migration is a blocker for the next release. I will now start reviewing your PRs :)

**Comment by atharvyadav22** [comment]
> > No issues they are not blocker to any of the work
> 
> @atharvyadav22 The migration work is very slow. So it needs to be done as a priority. Now, #4746 is almost done. The only `CoreReaderFragment` is left to refactor, which is a huge class and will take time. Migration is a blocker for the next release. I will now start reviewing your PRs :)

Okay, I am already ahead of my proposal so if these migration tests needed proper written Tests i can help with that side by side.

**Comment by MohitMaliFtechiz** [comment]
> Okay, I am already ahead of my proposal so if these migration tests needed proper written Tests i can help with that side by side.

Thank you. The last fragment is left to refactor. I will take care of that. You can continue to GSOC tickets :)

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+  @BeforeEach
+  fun setup() {
+    container = mockk<ZimReaderContainer>()
+  }
+
+  @Test
+  fun titleToUrl_whenAlreadyArticleUrl_returnsOriginalValue() {
+    val result = container.titleToUrl("A/Kiwix")
+
```
Comment (gouri-panda) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt]: @atharvyadav22 The function was not named properly. The test name says it returns the original value but the function actually passes it through unchanged only if it starts with "A/" . Thhe test value `"A/Kiwix" `passes that gate bu t the assertion checks that raw suffix `"A/Kiwix"` comes back not the content URL.

**Review by gouri-panda — DISMISSED** [review]
@atharvyadav22 @MohitMaliFtechiz Looks good to me! But i would add a little bit edge cases like `tittleToUrl` with a null tittle from `getPageUrlFromTitle`. No Unicode or special character tests for `contenntUrl`.

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    assertEquals(
+      "A/Kiwix",
+      result
+    )
+
+    verify(exactly = 0) {
+      container.getPageUrlFromTitle(any())
+    }
+  }
+
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt:54]: We should add a test for an empty title as well.

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@atharvyadav22 agree with @gouri-panda We should add those test cases.

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  @Test
+  fun titleToUrl_whenArticleDoesNotExist_returnsNull() {
+    every {
+      container.getPageUrlFromTitle("Unknown")
+    } returns null
+
+    val result = container.titleToUrl("Unknown")
+
+    assertNull(result)
+  }
```
Comment (atharvyadav22) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt:82]: @gouri-panda 
I have already added `null` case here.

Comment (MohitMaliFtechiz) [reply core/src/test/java/org/kiwix/kiwixmobile/core/utils/ZimReaderContainerUtilsTest.kt:82]: @atharvyadav22 Thanks for adding this. I have added the remaining edge cases.

**Review by MohitMaliFtechiz — APPROVED** [review]

**Review by gouri-panda — APPROVED** [review]
